PrebidMobile/Rendering/PrebidMobileRendering/PrebidMobileRendering/Prebid/PBMCacheRenderers/MediaView.swift

```diff
@@ -0,0 +1,288 @@
+    
+    func pauseWith(state: MediaViewState) {
+        guard state == .playing, let adViewManager = self.adViewManager  else {
+            return
+        }
+        self.state = state
+        adViewManager.pause()
+        delegate?.onMediaPlaybackPaused(self)
+    }
+    
+    @objc public func canAutoResume() -> Bool {
+        state == .pausedAuto
+    }
+    
+    @objc public func resume() {
+        guard isPaused, let adViewManager = self.adViewManager  else {
+            return
+        }
+        state = .playing
+        adViewManager.resume()
+        delegate?.onMediaPlaybackResumed(self)
+    }
+    
+    var isPaused: Bool { state == .pausedAuto || state == .pausedByUser }
+    var isActive: Bool { state == .playing || isPaused }
+    
+    // MARK: - PBMAdViewManagerDelegate protocol
+    
+    @objc public func viewControllerForModalPresentation() -> UIViewController? {
+        let mediaData = self.mediaData ?? mediaDataToLoad
+        let provider = mediaData?.nativeAdHooks.viewControllerProvider
+        return provider?() ?? UIViewController()
+    }
+    
+    @objc public func adLoaded(_ pbmAdDetails: PBMAdDetails) {
+        state = .playbackNotStarted
+        reportSuccess()
+    }
+
+    @objc public func failed(toLoad error: Error) {
+        reportFailureWithError(error, markLoadingStopped: true)
+    }
+
+    @objc public func adDidComplete() {
+        // FIXME: Implement
+    }
+
+    @objc public func videoAdDidFinish() {
+        state = .playbackFinished
+        delegate?.onMediaPlaybackFinished(self)
+    }
+
+    @objc public func videoAdWasMuted() {
+        delegate?.onMediaPlaybackMuted(self)
+    }
+
+    @objc public func videoAdWasUnmuted() {
+        delegate?.onMediaPlaybackUnmuted(self)
+    }
+
+    @objc public func adDidDisplay() {
+        // FIXME: Implement
+    }
+
+    @objc public func adWasClicked() {
+        // FIXME: Implement
+    }
+
+    @objc public func adViewWasClicked() {
+        // FIXME: Implement
+    }
+
+    @objc public func adDidExpand() {
+        // FIXME: Implement
+    }
+
+    @objc public func adDidCollapse() {
+        // FIXME: Implement
+    }
+
+    @objc public func adDidLeaveApp() {
+        // FIXME: Implement
+    }
+
+    @objc public func adClickthroughDidClose() {
+        // FIXME: Implement
+    }
+
+    @objc public func adDidClose() {
+        // FIXME: Implement
+    }
+
+    @objc public func displayView() -> UIView { self }
+    
+    // MARK: - Private Helpers
+
+    func reportFailureWithError(_ error: Error?, markLoadingStopped: Bool) {
+        if markLoadingStopped {
+            vastTransactionFactory = nil
+            mediaDataToLoad = nil
+        }
+    }
+
+    func reportSuccess() {
+        mediaData = mediaDataToLoad
+        vastTransactionFactory = nil
+        bindPlaybackToViewability = shouldBindPlaybackToViewability
+        delegate?.onMediaLoadingFinished(self)
+    }
+
+    func display(transaction: PBMTransaction?) {
```

Comment:
if this method can't work without transaction need to remove the optionality of the parameter. As a result, a caller will be responsible for validating the object.